config.go

```diff
@@ -69,29 +80,49 @@ func Parse(config map[string]string) (Config, error) {
 		return Config{}, fmt.Errorf("unknown payload format %q", config[Format])
 	}
 
+	// check if the PayloadFile parameter is compatible with other parameters
+	if parsed.PayloadFile != "" && parsed.Format == FormatStructured {
+		return Config{}, errors.New("payload file can only go with raw format")
+	}
+
+	// parse fields
+	fieldsMap, err := parseFieldsMap(config)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed parsing field spec: %w", err)
+	}
+
+	if len(fieldsMap) != 0 && parsed.PayloadFile != "" {
+		return Config{}, errors.New("cannot specify fields and payload field at the same time")
+	}
+	if len(fieldsMap) == 0 && parsed.PayloadFile == "" {
+		return Config{}, errors.New("either fields or a payload need to be specified")
+	}
```

Comment:
I can definitely make this shorter, but I believe it's more readable this way, and we can return a more accurate error message.